### PR TITLE
Add feature for rotating (automatically deleting) older backup sets

### PR DESCRIPTION
**Pull request by johnshearar**

### CODE REVIEW

File: lib/mongo_oplog_backup/rotate.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        MongoOplogBackup.log.info "Rotating out old backups completed."
+      end
+    end
+
+    # @param [Array<Pathname>] source_list List of Pathnames for the full backup sets in the backup directory.
+    # @returns [Array<Pathname]
+    def filter_for_deletion(source_list)
+      # The most recent dir might not be the active one (eg. if the mongodump fails)
+      source_list = source_list.reject { |path| path.basename.to_s == current_backup_name }
+      source_list = source_list.sort.reverse.drop(1) # Exclude the newest dir (which will be the current or previous backup)
```
Comment (rkistner) [inline lib/mongo_oplog_backup/rotate.rb]: Shouldn't this be `drop(KEEP_MINIMUM_SETS)`?

File: lib/mongo_oplog_backup/rotate.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    def timestamp_from_string(string)
+      match = /(\d+)(?::(\d+))?/.match(string)
+      return nil unless match
+      s1 = match[1].to_i
+      i1 = match[2].to_i
+      BSON::Timestamp.new(s1,i1)
+    end
+
+    def lock(lockname, &block)
+      File.open(lockname, File::RDWR|File::CREAT, 0644) do |file|
```
Comment (rkistner) [inline lib/mongo_oplog_backup/rotate.rb]: Can this code be shared with the backup process?

File: lib/mongo_oplog_backup/rotate.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        state_file = config.global_state_file
+        state = JSON.parse(File.read(state_file)) rescue nil
+        state ||= {}
+        @current_backup_name = state['backup']
+      end
+      @current_backup_name
+    end
+
+    def perform
+      lock(config.global_lock_file) do
```
Comment (rkistner) [inline lib/mongo_oplog_backup/rotate.rb]: Is the idea that this shouldn't run concurrently with backups?

Comment (johnshearar) [reply lib/mongo_oplog_backup/rotate.rb]: Correct, specifically not concurrent with Full backups to ensure that the state is stable. Perhaps overly cautious?

**Pull request merged**